run-cf-tests.sh

```diff
@@ -22,8 +22,13 @@ rm -rf /tmp/"$USER"/checker-framework
 /tmp/"$USER"/plume-scripts/git-clone-related typetools checker-framework /tmp/"$USER"/checker-framework
 export CHECKERFRAMEWORK=/tmp/"$USER"/checker-framework
 
-# see https://stackoverflow.com/questions/58033366/how-to-get-current-branch-within-github-actions/58035262
-dljc_branch_name=${GITHUB_REF##*/}
+if [[ "${BUILD_SOURCEBRANCHNAME+x}" ]]; then
```

Comment:
I think it should be possible to dispense with this logic and use `git-clone-related` below.
That ought to work on Azure Pipelines now, and the `github-actions` fork of `plume-scripts` may work for GitHub Actions too.
Does it not work on Azure Pipelines?

Reply:
The reason this logic is necessary is that there are implicitly two different uses of `git-clone-related` in this script. The first is the obvious one, which uses it to fetch the Checker Framework. The second use of `git-clone-related` is in the Checker Framework itself, as part of the `wpiManyTests` and `wpiPlumeLibTests` tasks, to get the do-like-javac repository - i.e., the repository being tested here.

This logic is needed for the second call to `git-clone-related` to find the right version of do-like-javac. Consider the case where we are testing a feature branch of do-like-javac, called `foo`. If there is no corresponding `foo` branch in the Checker Framework, then `git-clone-related` *in this script* will check out the master branch of `typetools/checker-framework`. This is desirable for testing, but it breaks the contract that `git-clone-related` *in the Checker Framework* expects, which is that if we want a particular branch of the other repository (in the case of calling from the CF, *this repository*!) we'll be on the corresponding branch. So we need to get the branch name explicitly and force the CF to be on that branch.

It's possible that `git-clone-related` does the right thing in this particular case even without this code, because it always fetches the name of the branch that it's cloning from CI information instead of the local git repository. As far as I can tell, that's an implementation detail that's not guaranteed - the documentation makes no mention of it - so I didn't want to rely on it.

Reply:
I think that `git-clone-related` does the right thing in this case, assuming that the two repositories are siblings (which is the default and is desirable).

Here is what I think you are concerned about:
 1. CI clones do-like-javac branch foo
 2. in do-like-javac branch foo, git-clone-related tries to clone checker-framework branch foo, but branch foo does not exist so it creates directory ../checker-framework which is on branch master
 3. in directory ../checker-framework, git-clone-related tries to clone do-like-javac branch master.  Wrong branch!

However, if the destination already exists, [`git-clone-related` uses it](https://github.com/plume-lib/plume-scripts/blob/master/git-clone-related#L23).  So the third step is actually:

 3. in directory ../checker-framework, git-clone-related tries to clone do-like-javac branch master.  It finds that ../do-like-javac already exists and uses it.  It's the version that CI cloned in step 1, and is on branch foo.

Does this address your concern?

Reply:
You understood my concern correctly.

However, this assumption you made is incorrect:
> assuming that the two repositories are siblings (which is the default and is desirable).

When the Checker Framework clones do-like-javac, it does so in its own directory tree (at `checker/bin/.do-like-javac`) rather than as a sibling of itself. This suggests a solution: we could modify the Checker Framework to clone do-like-javac as a sibling, which would remove the need for this code.